parser_test.go

```diff
@@ -13,57 +13,54 @@ func TestListCommands(t *testing.T) {
+	}{
+		"valid handler":     {&Cfg{handler: map[string]Handler{}}, "test", map[string]interface{}{"handler": "docker_compose_run", "service": "app",}, ""},
+		"invalid handler":   {&Cfg{}, "test", map[string]interface{}{"handler": "docker_compose_run"}, "error in strategy test: field service required but not set"},
+		"additional fields": {&Cfg{}, "foo", map[string]interface{}{"handler": "docker_compose_run", "service": "test", "other": "field"}, "error in strategy foo: additonal field(s) detected: other"},
 	}
-	err := cfg.generateHandler("test", settings)
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg.handler["test"])
-}
-
-func TestGenerateHandlerError(t *testing.T) {
-	cfg := &Cfg{}
-	err := cfg.generateHandler("test", map[string]interface{}{
-		"handler": "docker_compose_run",
-	})
-	assert.Error(t, err)
-	assert.Equal(t, err.Error(), "Error in strategy test: field service required but not set")
 
-	err = cfg.generateHandler("foo", map[string]interface{}{
-		"handler": "docker_compose_run",
-		"service": "test",
-		"other":   "field",
-	})
-	assert.Error(t, err)
-	assert.Equal(t, err.Error(), "Error in strategy foo: additonal field(s) detected: other")
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := test.cfg.generateHandler(test.handlerName, test.settings)
+			if test.expErr != "" {
+				assert.EqualError(t, err, test.expErr)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, test.cfg.handler[test.handlerName])
+			}
+		})
+	}
 }
 
-func TestGetHandlerFor(t *testing.T) {
-	var handler Handler
-	var err error
-
+func Test_GetHandlerFor(t *testing.T) {
 	cfg, err := GenerateConfig([]byte(fullYaml))
 	assert.NoError(t, err)
 
-	// Without strict mode for a known command
-	handler, err = cfg.GetHandlerFor("ls", false)
-	assert.NoError(t, err)
-	assert.NotNil(t, handler)
-
-	// With strict mode for an unkown command
-	handler, err = cfg.GetHandlerFor("some-cmd", true)
-	assert.Error(t, err, ErrUndefinedCommand)
-	assert.Nil(t, handler)
-
-	// Without strict mode for an unkown command
-	handler, err = cfg.GetHandlerFor("some-cmd", false)
-	assert.NoError(t, err)
-	assert.NotNil(t, handler)
+	tests := map[string]struct {
+		command    string
+		strictMode bool
+		expErr     error
+	}{
+		"known cmd":                    {"ls", false, nil},
+		"unknown cmd":                  {"some-cmd", false, nil},
+		"unknown cmd, strict":          {"some-cmd", true, ErrUndefinedCommand},
+		"custom cmd with path, strict": {"/bin/ls", true, nil},
+	}
 
-	// When specifying a path to the command
-	handler, err = cfg.GetHandlerFor("/bin/ls", true)
-	assert.NoError(t, err)
-	assert.NotNil(t, handler)
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			handler, err := cfg.GetHandlerFor(test.command, test.strictMode)
+			if test.expErr != nil {
+				assert.EqualError(t, err, test.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, handler)
+			}
+		})
+	}
```

Comment:
I have to get used to this style of testing - very hard to quickly scan and understand what the expected (failure) behavior is. Is it more common to put test cases into a struct/data structure instead of writing individual tests?